test/rubygems/test_gem_indexer.rb

```diff
@@ -31,15 +31,26 @@ def setup
     @indexer = Gem::Indexer.new(@indexerdir)
   end
 
+  def teardown
+    FileUtils.rm_rf(@indexer.directory)
+  ensure
+    super
+  end
```

Comment:
Here we have the same issue again about `@indexer` being potentially undefined when `teardown` is run. Since we don't have `around` hooks available, I think I would extract a `with_indexer(dir)` method that creates the proper directories, instantiates the index, yields to the passed block, and deletes the directory afterwards. And explicitly wrap every test that needs that with it.